src/Provider/AbstractProvider.php

```diff
@@ -115,6 +133,29 @@ public function getHttpClient()
         return $client;
     }
 
+    /**
+     * Set the instance of the CSPRNG random generator factory to use.
+     *
+     * @param  RandomFactory $factory
+     * @return $this
+     */
+    public function setRandomFactory(RandomFactory $factory)
```

Comment:
@SammyK this type hint will trigger an error if the parameter is not the correct type.